### PR TITLE
feat: add lint staged

### PR DESCRIPTION
**Pull request by Tanuj22**
Signed-off-by: Tanuj22 <tanujagarwal22@gmail.com>

**Description**
Add lint staged to run eslint only for files that are staged to be committed.

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -16,11 +16,14 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "npm run lint",
-      "pre-push": "npm run lint",
-      "...": "..."
```
Comment (kushthedude) [inline ui/package.json:21]: Add it for pre-push too.

Comment (Tanuj22) [reply ui/package.json:21]: It isn't supported for pre-push. An ongoing discussion here- https://github.com/okonet/lint-staged/issues/303

Comment (kushthedude) [reply ui/package.json:21]: Cool!

**Review by kushthedude — APPROVED** [review]

**Comment by kushthedude** [comment]
Congrats on your first PR in meshery @Tanuj22 😄

**Comment by leecalcote** [comment]
Thank you for this, @Tanuj22

**Pull request merged**